sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/CustomModelTransforms.java

```diff
@@ -0,0 +1,165 @@
+            trainingDocumentInfoList =
+                modelResponse.getTrainResult().getTrainingDocuments().stream()
+                    .map(trainingDocumentItem -> new TrainingDocumentInfo(
+                        trainingDocumentItem.getDocumentName(),
+                        TrainingStatus.fromString(trainingDocumentItem.getStatus().toString()),
+                        trainingDocumentItem.getPages(),
+                        transformTrainingErrors(trainingDocumentItem.getErrors())))
+                    .collect(Collectors.toList());
+            modelErrors = transformTrainingErrors(modelResponse.getTrainResult().getErrors());
+        }
+
+        List<CustomFormSubModel> subModelList = new ArrayList<>();
+        String formType = "form-";
+        // unlabeled model
+        if (modelResponse.getKeys() != null) {
+            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+            modelResponse.getKeys().getClusters().forEach((clusterKey, clusterFields) -> {
+                forEachWithIndex(clusterFields, (index, eachField) -> {
+                    String fieldName = "field-" + index;
+                    fieldMap.put(fieldName, new CustomFormModelField(eachField, fieldName, null));
+                });
+                subModelList.add(new CustomFormSubModel(
+                    null,
+                    fieldMap,
+                    formType + clusterKey));
+            });
+        } else if (modelResponse.getTrainResult() != null && modelResponse.getTrainResult().getFields() != null) {
+            // labeled model
+            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+            modelResponse.getTrainResult().getFields()
+                .forEach(formFieldsReport -> fieldMap.put(formFieldsReport.getFieldName(),
+                    new CustomFormModelField(null, formFieldsReport.getFieldName(),
+                        formFieldsReport.getAccuracy())));
+            subModelList.add(new CustomFormSubModel(
+                modelResponse.getTrainResult().getAverageModelAccuracy(),
+                fieldMap,
+                formType + modelInfo.getModelId()));
+        }
+
+        return new CustomFormModel(
+            modelInfo.getModelId().toString(),
+            CustomFormModelStatus.fromString(modelInfo.getStatus().toString()),
+            modelInfo.getCreatedDateTime(),
+            modelInfo.getLastUpdatedDateTime(),
+            new IterableStream<>(subModelList),
+            modelErrors,
+            new IterableStream<TrainingDocumentInfo>(trainingDocumentInfoList));
+    }
+
+    /**
+     * Helper method to convert the list of {@link ErrorInformation} to list of {@link FormRecognizerError}.
+     *
+     * @param trainingErrorList The list of {@link ErrorInformation}.
+     *
+     * @return The list of {@link FormRecognizerError}
+     */
+    private static IterableStream<FormRecognizerError> transformTrainingErrors(
```

Comment:
Prefer to use a List<> instead of IterableStream<>. List provides more convenient APIs to work with. Since the original `trainingErrorList` is already a list, streaming is not necessary.

It will also match your javadoc :)

Reply:
updated this to be a list.